src/main/java/core/basesyntax/factory/Factory.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax.factory;
+
+import core.basesyntax.dao.BetDaoImpl;
+import core.basesyntax.dao.UniversalDao;
+import core.basesyntax.dao.UserDaoImpl;
+import core.basesyntax.lib.Dao;
+import core.basesyntax.lib.NoDaoAnnotationInTheClass;
+
+public class Factory {
+    private static UniversalDao universalDao;
+
+    public static UniversalDao getBetDao() throws NoDaoAnnotationInTheClass {
+        if (!BetDaoImpl.class.isAnnotationPresent(Dao.class)) {
+            throw new NoDaoAnnotationInTheClass(message(BetDaoImpl.class));
+        }
```

Comment:
We don't need to do this check every time when getBetDao() is invoked - only when we are initializing universalDao field.